### PR TITLE
feat: add getInterval methods to the Solution classes for Sequence/Solution classes

### PR DESCRIPTION
**Pull request by vishwa2710**
## Release Notes

- **New Features**
	- Introduced `get_interval` method in both `Segment::Solution` and `Sequence::Solution` classes, allowing users to retrieve the time interval associated with segment and sequence solutions.

- **Bug Fixes**
	- Enhanced error handling in the new `get_interval` methods to manage empty states.

- **Tests**
	- Added new test classes and fixtures for `Segment.Solution` and `Sequence.Solution` to validate the functionality of the new interval retrieval methods, ensuring comprehensive coverage.

### CODE REVIEW

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/Sequence.cpp
Comment (apaletta3) [inline src/OpenSpaceToolkit/Astrodynamics/Trajectory/Sequence.cpp:1]: Could you also add a test to Sequence.getInterval as well?

File: bindings/python/test/trajectory/test_segment.py [hunk 1]
```diff
@@ -135,6 +135,64 @@ def thruster_dynamics() -> Thruster:
...
+        assert segment_solution.compute_delta_mass() is not None
+
+    # def test_extract_maneuvers(
+    #     self,
+    #     dynamics: list,
+    #     state: State,
+    # ):
+    #     segment_solution: Segment.Solution = Segment.Solution(
+    #         name="A Segment",
+    #         dynamics=dynamics,
```
Comment (apaletta3) [inline bindings/python/test/trajectory/test_segment.py:193]: What's this doing here? Will you implement it?

**Review by apaletta3 — APPROVED** [review]
Looks good, one comment to address

File: bindings/python/test/trajectory/test_segment.py (hunk 1 above)
Comment (vishwa2710) [reply bindings/python/test/trajectory/test_segment.py]: Done!

**Review by apaletta3 — APPROVED** [review]

**Pull request merged**